api/types.py

```diff
@@ -0,0 +1,9 @@
+from . import util
+
+# Origin represents the different methods a request can be authenticated.
+Origin = util.Enum('Origin', {
+    'user':    'user',    # An authenticated user
+    'device':  'device',  # A connected device (reaper, script, etc)
+    'job':     'job',     # Made on behalf of a job (downloading data, uploading results, etc)
+    'unknown': 'unknown', # Other or public
+})
```

Comment:
`Origin` is referenced all over the codebase, and it's not going in `base` because import cycles.

I want to start placing the other Enums and stuff (from jobs, etc) here.